### PR TITLE
PL-129934 use upstream gitlab & jemalloc

### PR DESCRIPTION
**Pull request by mkg20001**
@flyingcircusio/release-managers

## Release process

Impact:

Changelog:
- Use GitLab from upstream with registry patches (Note upstream PR for registry still pending) to replace puma with unicorn for memory usage
- Re-add fc changes via gitlab/override.nix file
- Use jemalloc to further reduce memory usage

## Security implications

- [x] [Security requirements](https://wiki.flyingcircus.io/System_Development_Guideline#Security_requirement_principles_and_testing) defined? (WHERE)
  - reduce amount of custom code -> use gitlab module from upstream
  - internal Gitlab components should listen to localhost only 
- [x] Security requirements tested? (EVIDENCE)
  - automated test still runs, manually tested on staging instance
  - checked listening ports on staging instance

### CODE REVIEW

File: nixos/roles/gitlab.nix [hunk 1]
```diff
@@ -50,13 +50,13 @@ in
...
       initialRootPasswordFile = "/srv/gitlab/secrets/root_password";
-      redisPassword = config.services.redis.requirePass;
+      redisUrl = "redis://:${config.services.redis.requirePass}@localhost:6379/";
       statePath = "/srv/gitlab/state";
       https = true;
       port = 443;
       host = cfg.hostName;
 
-      extraGitlabRb = ''
+      /* extraGitLabRb = ''
```
Comment (ctheune) [inline nixos/roles/gitlab.nix]: Remove unused code, no need to keep this around.

**Review by ctheune — COMMENTED** [review]
@dpausp Please do a proper review and will fill in the security implications. I have made a note for myself to contact @chrschm and get @mkg20001 bootstrapped with our ISMS requirements.

File: nixos/services/gitlab/default.nix [hunk 2]
```diff
@@ -224,7 +221,9 @@ in {
 
   imports = [
     (mkRenamedOptionModule [ "services" "gitlab" "stateDir" ] [ "services" "gitlab" "statePath" ])
+    (mkRenamedOptionModule [ "services" "gitlab" "backupPath" ] [ "services" "gitlab" "backup" "path" ])
     (mkRemovedOptionModule [ "services" "gitlab" "satelliteDir" ] "")
+    ./override.nix
```
Comment (dpausp) [inline nixos/services/gitlab/default.nix]: content from override.nix should be added to nixos/roles/gitlab.nix

File: nixos/services/default.nix [hunk 3]
```diff
@@ -20,6 +20,7 @@ in {
     ./collectdproxy.nix
     ./consul.nix
     ./gitlab
+    ./gitlab/override.nix
```
Comment (dpausp) [inline nixos/services/default.nix]: content from override.nix should be added to nixos/roles/gitlab.nix

**Review by dpausp — CHANGES_REQUESTED** [review]
nixos/services/gitlab/defaultUnicornConfig.rb can be removed

File: nixos/roles/gitlab.nix [hunk 4]
```diff
@@ -44,28 +44,55 @@ in
       (writeScriptBin "gitlab-show-config" ''jq < /srv/gitlab/state/config/gitlab.yml'')
     ];
 
+    # all logs to /var/log
+    systemd.tmpfiles.rules = [
```
Comment (dpausp) [inline nixos/roles/gitlab.nix:48]: cfg is wrong here, cfg is defined as config.flyingcircus.roles.gitlab in the role

you want config.services.gitlab here

**Review by dpausp — CHANGES_REQUESTED** [review]
build breaks at the moment, please rebase after the fix and run the gitlab test

**Pull request merged**